src/components/NewsletterSignup/NewsletterSignup.tsx

```diff
@@ -0,0 +1,227 @@
+import {
+  Box,
+  chakra,
+  Stack,
+  useColorModeValue,
+  useStyleConfig,
+  VStack,
+} from "@chakra-ui/react";
+import React, { forwardRef, useRef, useState } from "react";
+
+import Button from "../Button/Button";
+import Form, { FormField } from "../Form/Form";
+import Icon from "../Icons/Icon";
+import Link from "../Link/Link";
+import Text from "../Text/Text";
+import TextInput from "../TextInput/TextInput";
+import Heading from "../Heading/Heading";
+import useNYPLBreakpoints from "../../hooks/useNYPLBreakpoints";
+import { getSectionColors } from "../../helpers/getSectionColors";
+import { SectionTypes } from "../../helpers/types";
+
+export const newsletterSignupViewTypeArray = [
+  "form",
+  "confirmation",
+  "error",
+] as const;
+
+export type NewsletterSignupViewType =
+  typeof newsletterSignupViewTypeArray[number];
+
+interface NewsletterSignupProps {
+  /** Additional class name to add. */
+  className?: string;
+  /** Used to add additional information to the default confirmation message in
+   * the confirmation view. */
+  confirmationText?: string | JSX.Element;
+  /** Used to add description text above the form input fields in
+   * the initial/form view. Accepts a string or a JSX */
+  descriptionText?: string | JSX.Element;
+  /** Optional: Used to populate the Text component rendered below the input field. */
+  formHelperText?: string | JSX.Element;
+  /** A data object containing key/value pairs that will be added to the form
+   * field submitted data. */
+  hiddenFields?: any;
+  /** ID that other components can cross-reference for accessibility purposes */
+  id?: string;
+  /** Toggles the invalid state for the email field. */
+  isInvalidEmail?: boolean;
+  /* Optional value to determine the section color highlight */
+  newsletterSignupType?: SectionTypes;
+  /** A callback handler function that will be called when the form is submitted.
+   * The function provided must expect an object of key:value pairs. This object will
+   * include {email}, being the user's input, plus any objects passed via the hiddenValues prop.
+   */
+  onSubmit: (values: { [key: string]: string }) => any;
+  /** Used to populate the `<h3>` header title. */
+  title: string;
+  /** Used to specify what is displayed in the component content area. */
+  view?: NewsletterSignupViewType;
+}
+
+/**
+ * The NewsletterSignup component provides a way for patrons to register for an
+ * email-based newsletter distribution list.
+ *
+ * The component can show four different views, depending on the state of the
+ * email submission: form, loading, confirmation, and error.
+ */
+export const NewsletterSignup = chakra(
+  forwardRef<any, NewsletterSignupProps>(
+    (
+      {
+        className,
+        confirmationText = "Thank you! You have successfully subscribed to our email updates! You can update your email subscription preferences at any time using the links at the bottom of the email.",
```

Comment:
Nitpick: can these longer sentences be constant variables assigned outside the component? Such as `const defaultConfirmationText = "Thank you.....";`

Reply:
Done